python/generator_utils.py

```diff
@@ -86,6 +86,40 @@ def __str__(self):
 
 class MemberVariable:
   """Simple class to hold information about a member variable"""
+
+  def parse_julia_type(self, **kwargs):
+    """Parse the given c++ type to a Julia type"""
+    self.cpp_type = kwargs.pop('cpp_type', None)
+    is_array = kwargs.pop('is_array', False)
+
+    builtin_types_map = {"int": "Int32", "float": "Float32", "double": "Float64",
+                         "bool": "Bool", "long": "Int32", "unsigned int": "UInt32",
+                         "unsigned long": "UInt32", "char": "Char", "short": "Int16",
+                         "long long": "Int64", "unsigned long long": "UInt64", "std::string": "String"}
+
+    if self.cpp_type in BUILTIN_TYPES:
+      return builtin_types_map[self.cpp_type]
+
+    if not is_array:
+      if self.cpp_type.startswith('std::'):
+        self.cpp_type = self.cpp_type[5:]
+      if self.cpp_type in ALLOWED_FIXED_WIDTH_TYPES:
+        if ALL_FIXED_WIDTH_TYPES_RGX.match(self.cpp_type):
+          regex_string = re.search("(8|16|32|64)", self.cpp_type).group()
+          if self.cpp_type.startswith('u'):
+            self.cpp_type = ("UInt" + regex_string)
+          elif self.cpp_type.startswith('i'):
+            self.cpp_type = ("Int" + regex_string)
+          return self.cpp_type
```

Comment:
This could potentially be simplified by adapting the `ALL_FIXED_WIDTH_TYPES_RGX`. In principle the number of bits is already inside a capture group, so it should available. Looking at the definition, only the `u` is not yet inside a capture group.